.github/workflows/all-pr-checks.yml

```diff
@@ -0,0 +1,66 @@
+---
+name: Test All
+on:
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+jobs:
+  # This job checks for the changed paths
+  changes:
+    name: Get changed packages
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      packages: ${{ steps.filter.outputs.changes }}
+    steps:
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          # We can codify the path dependencies between the packages here
```

Comment:
**suggestion**: it seems this steps produces outputs we depend on later on. Could you add a comment what outputs we expect, and maybe provide example values? 

The [action's documentation](https://github.com/dorny/paths-filter?tab=readme-ov-file#outputs) mentions "filters" and "changes", but we also use `changes` as a GH job name, and `filter` as a step ID, and this naming overlap made it difficult for me to understand our set up.

Reply:
does this help? 4467194

Reply:
Ohhh, I can see it now. Yes, it's much easier now, thanks!